attest/tpm_windows.go

```diff
@@ -410,42 +304,73 @@ func (t *TPM) MintAIK(opts *MintOptions) (*Key, error) {
 		return nil, fmt.Errorf("pcp failed to read attestation key properties: %v", err)
 	}
 
-	return &Key{
-		hnd:               kh,
-		KeyEncoding:       KeyEncodingOSManaged,
-		TPMVersion:        t.version,
-		Purpose:           AttestationKey,
-		PCPKeyName:        name,
-		Public:            props.RawPublic,
-		CreateData:        props.RawCreationData,
-		CreateAttestation: props.RawAttest,
-		CreateSignature:   props.RawSignature,
-	}, nil
-}
+	switch t.version {
+	case TPMVersion12:
+		return &AIK{
+			aik: &key12{
+				hnd:        kh,
+				pcpKeyName: name,
+				public:     props.RawPublic,
+			},
+		}, nil
 
-// LoadKey loads a previously-created key into the TPM for use.
-// A key loaded via this function needs to be closed with .Close().
-func (t *TPM) LoadKey(opaqueBlob []byte) (*Key, error) {
-	var k Key
-	var err error
-	if err = json.Unmarshal(opaqueBlob, &k); err != nil {
-		return nil, fmt.Errorf("Unmarshal failed: %v", err)
-	}
+	case TPMVersion20:
+		return &AIK{
+			aik: &key20{
+				hnd:               kh,
+				pcpKeyName:        name,
+				public:            props.RawPublic,
+				createData:        props.RawCreationData,
+				createAttestation: props.RawAttest,
+				createSignature:   props.RawSignature,
+			},
+		}, nil
 
-	if k.TPMVersion != t.version {
-		return nil, errors.New("key TPM version does not match opened TPM")
+	default:
+		return nil, fmt.Errorf("cannot handle TPM version: %v", t.version)
 	}
-	if k.KeyEncoding != KeyEncodingOSManaged {
-		return nil, fmt.Errorf("unsupported key encoding: %x", k.KeyEncoding)
+}
+
+// LoadAIK loads a previously-created AIK into the TPM for use.
+// A key loaded via this function needs to be closed with .Close().
+func (t *TPM) LoadAIK(opaqueBlob []byte) (*AIK, error) {
```

Comment:
nit: can you make this exported function defined in a file without any build tags? Then immediately call out to a private method that each arch implements?

```
func (t *TPM) LoadAIK(opaqueBlob []byte) (*AIK, error) {
   return t.loadAIK(opaqueBlob)
}
```

That way you can write the godoc once and it won't get out of sync